Ale isd gen

jrcain-usgs opened the pull request:
## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+
+    Generate Instrument Support Data (ISD) for an image with ALE's `isd_generate.py`
+
+## Prerequisites
+
+You will need three things before you can generate an ISD from an image:
+
+- [Installation of ISIS](../../how-to-guides/environment-setup-and-maintenance/installing-isis-via-anaconda.md)
```
These prefers should be more in the form of a numbered list to show you do these in order.

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+    - [Locate an Image](../../getting-started/using-isis-first-steps/locating-and-ingesting-image-data.md)
+    - [Ingest the Image into ISIS](../../getting-started/using-isis-first-steps/locating-and-ingesting-image-data.md#introduction-to-importing)
+    - Add Spice Data to the image with [`spiceinit`](https://isis.astrogeology.usgs.gov/Application/presentation/Tabbed/spiceinit/spiceinit.html)*
+    - Download [SPICE data](https://naif.jpl.nasa.gov/naif/data.html) and set `$ALESPICEROOT`*
```
This should be higher on the list.

jrcain-usgs replied:
Do you always need NAIF Data/`$ALESPICEROOT`, or just for NaifSpice drivers?  (and for IsisSpice drivers, would you have to use `spiceinit` instead?

Kelvinrr replied:
ISISSPICE does not need kernels, anything with naif spice would. Most people would be using naif spice realistically.

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
````diff
@@ -0,0 +1,147 @@
+isd_generate [--arguments] [your-isis-cube-name].cub
+```
+
+To run isd_generate with no arguments on a cub named M1229488338LE.cub:
+
+```sh
+isd_generate M1229488338LE.cub
````
We should provide the cube

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+
+## Arguments
+
+### Output File
+
+`-o`, `--out` [filename]
+
+:   Optional output file.  If not specified, this will be set to 
+    the input filename with its final suffix replaced with .json. 
+    If multiple input files are provided, this option will be ignored 
+    and the default strategy of replacing their final suffix with 
+    .json will be used to generate the output file paths.
+
+`-c`, `--compress`
+
+:   Output a compressed isd json file with .br file extension. 
+    Ale uses the brotli compression algorithm. 
+    To decompress an isd file run: 
+    `python -c "import ale.isd_generate as isdg; isdg.decompress_json('/path/to/isd.br')"`
```
Breaking down the arguments doesn't make sense unless we are going to have runnable examples. Otherwise we should just tell them to run -h for parameters as these can change.

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+
+### Spice Data
+
+`-k`, `--kernel` [filename]
+
+:   Typically this is an optional metakernel file, care should be 
+    taken by the user that it is applicable to all of the input 
+    files.  It can also be a single ISIS cube, which is sometimes 
+    needed if the input file is a label file.
+
```
demonstrating this with a meta kernel seems like it would be a good working example and a tactile way of understanding kernels.

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+    files.  It can also be a single ISIS cube, which is sometimes 
+    needed if the input file is a label file.
+
+`-i`, `--only_isis_spice`
+
+:   Only use drivers that read from spiceinit'd ISIS cubes.
+
+`-n`, `--only_naif_spice`
+
+:   Only use drivers that generate fresh spice data.
+
```
examples for these would be good as well, this can show how the different drivers support the same file.

Kelvinrr replied:
if you have a spiceinited cube, and want to re-spiceinit, you can run with `--only-naif-spice`. `--only-isis-spice` if you don't want to risk re-spiceiniting. This is because of the fact that which driver it picks up first (the naif or isis one), is not guaranteed.

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+
+### Spatial Settings
+
+`-l`, `--local`
+
+:   Generate local spice data, an isd that is unaware of itself relative
+    to target body. This is largely used for landed/rover data.
+
+`-N`, `--nadir`
+
+:   Generate nadir spice pointing, an isd that has pointing 
+    directly towards the center of the target body.
```
having concrete examples of these would also be useful. Adam can help find an example for `--local`.

jrcain-usgs commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+
+You will need three things before you can generate an ISD from an image:
+
+- [Installation of ISIS](../../how-to-guides/environment-setup-and-maintenance/installing-isis-via-anaconda.md)
```
Do people need ISIS most of the time?  Like this table, or not quite?

|            | IsisLabel  | Pds3Label          |
|------------|------------|--------------------|
| IsisSpice  | Needs ISIS | Needs ISIS         |
| Naif Spice | Needs ISIS | Does not Need ISIS |

I'm still a bit vague on the finding of images. Are people going look for images based on what works in ALE, or do people have images already?

How/where do they get a PDS (Pds3Label in ALE drivers?) image vs an ISIS image, and do I need to tell them about how to find those here?

Kelvinrr replied:
tutorials should be followable with data, there are probably images you can use that already on the websites data dirtectory https://github.com/DOI-USGS/asc-public-docs/tree/main/docs/getting-started/data/image_to_ground

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,147 @@
+- An Image in ISIS format with SPICE data:
+    - [Locate an Image](../../getting-started/using-isis-first-steps/locating-and-ingesting-image-data.md)
+    - [Ingest the Image into ISIS](../../getting-started/using-isis-first-steps/locating-and-ingesting-image-data.md#introduction-to-importing)
+    - Add Spice Data to the image with [`spiceinit`](https://isis.astrogeology.usgs.gov/Application/presentation/Tabbed/spiceinit/spiceinit.html)*
```
only for isisspice yeah, but it's important that spiceinited cubes can also be re-spiceinited using isd_generate. If you were to run spiceinit again, it's simply using ALE in the background.

jrcain-usgs replied:
based on your feedback, * is not clear enough, reformatted.

jrcain-usgs commented:
Added examples for output, compression, verbose, kernels, naif-only.  Still need to talk to Adam about --local example.  Added section about setting up NAIF Data, could break out this section to it's own page later if we want.

Kelvinrr commented on docs/getting-started/using-ale/isd-generate.md:
```diff
@@ -0,0 +1,356 @@
+
+!!! warning ""
+
+    [[NAIF SPICE Data Home](https://naif.jpl.nasa.gov/naif/data.html)] 
+    [[Getting Spice Kernels from the NAIF Server](https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/Tutorials/pdf/individual_docs/34_naif_server.pdf)]
+
+    If you have enough space, NAIF recommends downloading the entire dataset for your spacecraft, detailed below.
+    Otherwise, you can try subsetting the data as detailed in the [NAIF Server tutorial](https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/Tutorials/pdf/individual_docs/34_naif_server.pdf).
+    You will need `wget`, a common download utility.
+
+    Let's download the Clementine archive as a test.
+
+    1.  Find your mission on the [PDS SPICE Archives](https://naif.jpl.nasa.gov/naif/data_archived.html) and copy the **Archive Link**.
+
+        -  The url for the Clementine archive is:    
+           `https://naif.jpl.nasa.gov/pub/naif/pds/data/clem1-l-spice-6-v1.0/clsp_1000`
+
+    1.  `cd` into you spice folder, the folder you have or will set as `$ALESPICEROOT`.
+
+    1.  Piece together the `wget` command.
+        1.  Starting with this base command... (Don't run it yet!)    
+            `wget -m -nH --cut-dirs=5 -R 'index.html*'`
+
+        1.  add the url...  
+            `-nv https://naif.jpl.nasa.gov/pub/naif/pds/data/clem1-l-spice-6-v1.0/`
```
We should link to the instructions on ISIS readme on how to download isis data using downloadIsisData.py
